### PR TITLE
Augmenting the `GetSchema` RPC to also work for `Table` and `All` type of input

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR reworks the `GetSchema` RPC to use the schema tracking offered by the schema-engine. This was introduced recently in the changes in #13121. We also augment the RPC to also work with `table` and `all` type instead of the `Views` type that it supported originally.

As part of this change, I have also renamed the schema_engine tables to have a more suitable name. Since they were added just a few days ago and never made to any release, this rename is safe and has no backward-compatibility issues. 

The `views` table in the health-streamer schema tracking is removed since it would no longer be used. Coincidentally, the new table in `schema_engine` has been called `views` too and they have a similar schema barring a single column which the sidecardb will take care off, during upgrades.

I also found a couple of bugs in the `GetSchema` RPC related to filtering by views and handling of multiple returned packets from the RPC. They have also been fixed as part of this endeavour.

## Related Issue(s)

- Prerequisite for #12967 

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vttablet/endtoend/framework/client.go [hunk 1]
```diff
@@ -415,9 +415,11 @@ func (client *QueryClient) UpdateContext(ctx context.Context) {
 }
 
 func (client *QueryClient) GetSchema(tableType querypb.SchemaTableType, tableNames ...string) (map[string]string, error) {
-	schemaDef := map[string]string{}
+	schemaDef := make(map[string]string)
 	err := client.server.GetSchema(client.ctx, client.target, tableType, tableNames, func(schemaRes *querypb.GetSchemaResponse) error {
-		schemaDef = schemaRes.TableDefinition
+		for tableName, schemaDefinition := range schemaRes.TableDefinition {
+			schemaDef[tableName] = schemaDefinition
+		}
```
Comment (GuptaManan100) [inline go/vt/vttablet/endtoend/framework/client.go:422]: This was another bug I found! We shouldn't just be overriding the map we get back, but we should update it.

**Comment by GuptaManan100** [comment]
`TestNewView` and `TestViewAndTable` are running as part of the `upgrade downgrade query serving schema` test, so we already have a test which verifies that the schema tracking works with v16 vtgate with the latest version of vttablet with the changes in this PR!

**Comment by GuptaManan100** [comment]
The test performance has also improved from running in `5.83 seconds` with v16 vttablet, versus running in `1.11 seconds` now. This is because the schema engine schema reload runs just as the ddl is executed while the older health-streamer one was on a timer tick.

File: go/vt/vttablet/tabletserver/schema/db.go [hunk 2]
```diff
@@ -372,3 +390,72 @@ func reloadDataInDB(ctx context.Context, conn *connpool.DBConn, altered []*Table
 	}
 	return nil
 }
+
+// GetFetchViewQuery gets the fetch query to run for getting the listed views. If no views are provided, then all the views are fetched.
+func GetFetchViewQuery(viewNames []string) (string, error) {
```
Comment (harshit-gangal) [inline go/vt/vttablet/tabletserver/schema/db.go:395]: all 3 methods look identical, may be they can be placed in a common function.

**Review by harshit-gangal — APPROVED** [review]

File: go/vt/vttablet/tabletserver/schema/db.go (hunk 2 above)
Comment (GuptaManan100) [reply go/vt/vttablet/tabletserver/schema/db.go:395]: I could do that, but then they'll have a bunch of switch cases, which isn't great either.

File: go/vt/vttablet/endtoend/rpc_test.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
+package endtoend
```
Comment (systay) [inline go/vt/vttablet/endtoend/rpc_test.go:1]: license header missing

**Review by systay — APPROVED** [review]
LGTM!

File: go/vt/vttablet/endtoend/rpc_test.go (hunk 3 above)
Comment (GuptaManan100) [reply go/vt/vttablet/endtoend/rpc_test.go:17]: Fixed.

**Pull request merged**